x-pack/plugins/maps/public/connected_components/toolbar_overlay/index.ts

```diff
@@ -6,13 +6,20 @@
  */
 
 import { connect } from 'react-redux';
+import { ToolbarOverlay } from './toolbar_overlay';
 import { MapStoreState } from '../../reducers/store';
+import { getDrawMode } from '../../selectors/ui_selectors';
 import { getGeoFieldNames } from '../../selectors/map_selectors';
-import { ToolbarOverlay } from './toolbar_overlay';
+import { DRAW_MODE } from '../../../common';
 
 function mapStateToProps(state: MapStoreState) {
+  const shapeDrawModeActive = getDrawMode(state) === DRAW_MODE.DRAW_SHAPES;
+  const pointDrawModeActive = getDrawMode(state) === DRAW_MODE.DRAW_POINTS;
   return {
-    showToolsControl: getGeoFieldNames(state).length !== 0,
+    showToolsControl:
+      getGeoFieldNames(state).length !== 0 && !(shapeDrawModeActive || pointDrawModeActive),
```

Comment:
I am not sure we want to hide the tools control while in edit mode. How about disabling the button or just keeping it there.

Reply:
I opted for disabling. Honestly I'd prefer to go in the opposite direction of stripping out unneeded controls, minimizing the TOC, etc. But that can be a separate discussion and PR. Minimally we don't want people messing with filters while drawing. It's a bad UX since it will kill draw mode as soon as filter mode is activated.